### PR TITLE
node: Fix incorrect LiveView fields

### PR DESCRIPTION
**Pull request by erikd**

### CODE REVIEW

File: cardano-node/cardano-node.cabal [hunk 1]
```diff
@@ -64,16 +64,20 @@ library
                      , network-mux
                      , optparse-applicative
                      , ouroboros-consensus
+                     , ouroboros-consensus-byron
+                     , ouroboros-consensus-shelley
                      , ouroboros-consensus-cardano
                      , ouroboros-network
+                     , ouroboros-network-framework
                      , process
+                     , shelley-spec-ledger
```
Comment (mrBliss) [inline cardano-node/cardano-node.cabal:73]: It's unfortunate that we have to list these dependencies here. The goal was that `ouroboros-consensus-cardano` would be enough for `cardano-node`.

I suppose an alternative would be to let `ouroboros-consensus-cardano` expose more.

File: cardano-node/src/Cardano/Node/TUI/LiveView.hs [hunk 2]
```diff
@@ -572,32 +576,64 @@ initLiveViewState = do
...
+        -- This is correct for Byron and Shelley.
+        MempoolCapacityBytes mempoolCapacity <- STM.atomically $ getCapacity (getMempool nodeKern)
+
+        currentLedger <- STM.atomically $ getCurrentLedger (getChainDB nodeKern)
+
+        pure lvs
+              { lvsNodeKernel = SJust (LVNodeKernel nodeKern)
+              , lvsMempoolCapacity =
+                  fromIntegral $
+                    mempoolCapacity `div` getMaxTxSize currentLedger
```
Comment (mrBliss) [inline cardano-node/src/Cardano/Node/TUI/LiveView.hs:634]: So this is a conservative estimate, I see.

File: cardano-config/src/Cardano/Config/Types.hs [hunk 3]
```diff
@@ -350,10 +368,23 @@ instance ToJSON NodeHostAddress where
...
+class HasTxMaxSize ledgerState where
+  getMaxTxSize :: ledgerState -> Word32
+
+instance HasTxMaxSize (ExtLedgerState (ShelleyBlock c)) where
+  getMaxTxSize = fromIntegral . _maxTxSize . getPParams . shelleyState . ledgerState
+
+instance HasTxMaxSize (ExtLedgerState ByronBlock) where
+  getMaxTxSize =
+    fromIntegral . ppMaxBlockSize . adoptedProtocolParameters
+      . cvsUpdateState . byronLedgerState . ledgerState
```
Comment (mrBliss) [inline cardano-config/src/Cardano/Config/Types.hs:384]: :+1:

File: cardano-node/src/Cardano/Node/TUI/LiveView.hs (hunk 2 above)
Comment (erikd) [reply cardano-node/src/Cardano/Node/TUI/LiveView.hs:634]: Yes, I understand. This is how the code esitmated the capacity based on hard coded values before.

File: cardano-node/cardano-node.cabal (hunk 1 above)
Comment (erikd) [reply cardano-node/cardano-node.cabal:73]: One possible solution would be move my `HasTxMaxSize` typeclass into `ouroboros-consensus-cardano`.

Comment (dcoutts) [reply cardano-node/cardano-node.cabal:73]: We could have a class of ledger state functions that overlap between Byron and Shelley and put that in `ouroboros-consensus-cardano`. This example is just getting one of the protocol params (tx size) that has an instance for both ledgers.

Comment (mrBliss) [reply cardano-node/cardano-node.cabal:73]: Yes, I expect more such type class will be needed in the future. Maybe there are already more such type classes in disguise in `cardano-node`?

Oh, I just realised: `RunNode` is *the* type class for such cross-block details. I think we should make `getMaxTxSize` part of `RunNode`.

**Comment by Jimbo4350** [comment]
bors merge

File: cardano-node/cardano-node.cabal (hunk 1 above)
Comment (erikd) [reply cardano-node/cardano-node.cabal:73]: I did suggest that, but to get things working more quickly we went with this solution,

Comment (intricate) [reply cardano-node/cardano-node.cabal:73]: I agree that it should be part of `RunNode`. We can revisit this and do that at some point.

Comment (mrBliss) [reply cardano-node/cardano-node.cabal:73]: https://github.com/input-output-hk/ouroboros-network/issues/2106

**Pull request merged**